lib/remote-config.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const request = require('request');
+
+class RemoteConfig {
+  constructor(url, refreshTimeSeconds) {
+    this.url = url;
+    this.refreshable = (refreshTimeSeconds) ? true : false;
+    this.refreshTimeSeconds = (refreshTimeSeconds) ? refreshTimeSeconds : 0;
+    this.isRefreshing = false;
+    this.nextRefresh = null;
+    this.configObject = null;
+  }
+
+  getConfigObject(context) {
+    return this.refreshCache(context);
+  }
+	
+  getConfigObjectFromSource(context) {
+    return new Promise((resolve, reject) => {
+      if (!this.url) {
+        return reject(new Error('Missing remote config URL.'));
+      }
+      return request(this.url, (error, response, body) => {
+        if (error) {
+          context.logger.error({ err: error }, `Error calling url ${this.url}.`);
+          return reject(`Error calling url ${this.url}.`);
```

Comment:
There is no need to log here, because `refreshCache()` already has a `catch` which logs the error, so just reject it.
Remember to reject an `Error` or an object, but not just a string.